clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -287,6 +287,7 @@ private KafkaProducer(ProducerConfig config, Serializer<K> keySerializer, Serial
             this.sender = new Sender(client,
                     this.metadata,
                     this.accumulator,
+                    config.getInt(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION) == 1,
```

Comment:
@hachikuji @jjkoshy I actually started to prefer the current patch when trying to update the patch. It looks a little weird to keep track of the in flight batches per partition which no one cares about unless the max in flight request is one.  So we are paying the extra tracking overhead but don't gain much.